src/components/Participant/RegistrationProcess/ContinueRegistration/Forms/CheckAnswersForm.tsx

```diff
@@ -34,7 +34,9 @@ function CheckAnswersForm(props: CheckAnswersFormProps) {
 
   return (
     <>
-      <DTEHeader as="h1">{content["register2-check-answers-header"]}</DTEHeader>
+      <DTEHeader as="h1" captionKey="register2-check-answers-header">
```

Comment:
I would imagine this functionality to be:

```
<DTEHeader as="h1" contentKey="register2-check-answers-header"></DTEHeader>
```

renders something like:

```
<DTEHeaderCaption contentKey={`${contentKey}-caption`} />
<StyledHeader {...props}>{content[contentKey]}</StyledHeader>
```

So that `contentKey` is specified only once and reused to both build the caption key and retrieve the header content itself so the header doesn't need to have a child element.

However an optional direct child on `DTEHeader` could either override or append to the CMS content. What would the preference be here?

Reply:
@con-nihr totally agree that this would be the best course of action however due to the recursive contentful modeller which accepts additional nodes as children I'm not sure if there is a way to achieve this.  Had looked at defining the key as a constant string and passing in but this was probably overkill. 

Happy to take feedback/suggestions if you can think of a better way to do this

This [file](https://github.com/PA-NIHR-CRN/dte/blob/main/src/Helper/contenful/modeller.tsx) basically takes the contentful node tree and traverses it recursively to add the custom DTE styles to any contentful content

Reply:
Recursively render `DTEHeader`?

```
<DTEHeader as="h1" captionKey="register2-check-answers-header" />
```

renders

```
if(captionKey){
    <DTEHeaderCaption contentKey={`${captionKey}-caption`} />
    <DTEHeader as="h1">{content[captionKey]}</DTEHeader>
} else{
    ... render DTEHeader normally
}
```

Reply:
It's when the content is passed in as rich text from contentful, `h1` tags may be added which wrap in the correct styling through the recursion.  Through the majority of the application the `DTEHeader` is manually added(and wrapped) using a dictionary to access the content however there are some instances(such as the cookie page) where this is not the case. 

The main bulk of the headers have been manually wrapped to ensure consistent styling/accessibility but in some cases the client wanted more control/flexibility.  This will be extended when the new content pages are brought in also

Also the conditional rendering of the Caption is handled within the `DTEHeaderCaption` by returning null if it does not exist

Reply:
Why would the extra level of recursion above not support those scenarios?

https://github.com/PA-NIHR-CRN/dte/pull/124#discussion_r1557325655